include/s3select_functions.h

```diff
@@ -643,6 +645,22 @@ struct _fn_substr : public base_function{
     
 };
 
+struct _fn_charlength : public base_function {
+
+    value v_str;
+ 
+    bool operator()(std::vector<base_statement*> * args,variable * result)
+    {
+        std::vector<base_statement*>::iterator iter = args->begin();
+        base_statement* str =  *iter;
+        v_str = str->eval();
```

Comment:
operator() is a highly intensive method.
u correctly declare v_str , out of operator() scope (otherwise it construct/destruct per each call).

can u avoid using assign operator (=) , prior to strlen?  it will save some instructions.